tests/test_checkout_caching.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# coding: utf-8
+import os
+import sys
+import unittest
+import tempfile
+from google.cloud import storage
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from dss import Config, Replica, BucketConfig
+from dss.stepfunctions import s3copyclient, gscopyclient
+from tests import infra
+from tests.infra import DSSAssertMixin, DSSUploadMixin, testmode
+from tests.infra.server import ThreadedLocalServer
+
+
+class TestCheckoutApi(unittest.TestCase, DSSAssertMixin, DSSUploadMixin):
+    test_bundle_uploaded: bool = False
+    bundle_uuid: str
+    bundle_version: str
+    file_uuid: str
+    file_version: str
+
+    class SpoofContext:
+        @staticmethod
+        def get_remaining_time_in_millis():
+            return 2000
+
+    @classmethod
+    def setUpClass(cls):
+        cls.app = ThreadedLocalServer()
+        cls.app.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.app.shutdown()
+
+    def setUp(self):
+        Config.set_config(BucketConfig.TEST)
+
+    @testmode.standalone
+    def test_aws_uncached_checkout_creates_tag(self):
```

Comment:
Please add doc strings to your test that follow "behavior when conditions" statements. For example "A files is tagged with 'uncached', when checked out from an aws bucket."

Reply:
I do appreciate the informative test names 👍

Reply:
Will do!  Thanks!